Add simplified default type

vlopes11 opened the pull request:
The annotation is an internal requirement of Microkelvin and should not
reflect the API as a requirement.

The maps should provide a simplified `K -> V` data structure.

Resolves #6

CPerezz commented on src/lib.rs:
```diff
@@ -16,19 +16,22 @@ use microkelvin::{
+pub struct KelvinHamt<K, V, A>([Bucket<K, V, A>; 4]);
 
-impl<K, V, A> Compound<A> for Hamt<K, V, A>
+pub type Hamt<K, V> = KelvinHamt<K, V, ()>;
```
Maybe add some doc comments since the type is exported.

vlopes11 replied:
Agreed, but its out the scope of the PR. Issue added to keep track:

https://github.com/dusk-network/dusk-hamt/issues/8

CPerezz reviewed (commented):
Just a suggestion

krl commented:
I would prefer you add a DefaultHamt if you need it, i don't see the point of just renaming a bunch of types, which itself is a breaking change.

`type DefaultHamt<K, V> = Hamt<K, V, ()>` would suffice and is a one-liner.

vlopes11 committed:
Add simplified default type

The annotation is an internal requirement of Microkelvin and should not
reflect the API as a requirement.

The maps should provide a simplified `K -> V` data structure.

Resolves #6
